File: launchschool_exercises/210_small_problems/strings_and_text_processing/swap-case.js
/*
  Write a function that takes a string as an argument, and returns that string with every lowercase letter changed to uppercase and every uppercase letter changed to lowercase. Leave all other characters unchanged.
*/

function swapcase(str) {
  return str.split('').map(function (ch) {
    if (isUpper(ch)) {
      return ch.toLowerCase();
    } else if (isLower(ch)) {
      return ch.toUpperCase(ch);
    } else {
      return ch;
    }
  }).join('');
}

function isUpper(ch) {
  return /[A-Z]/.test(ch);
}

function isLower(ch) {
  return /[a-z]/.test(ch);
}

console.log(swapcase('CamelCase'));              // "cAMELcASE"
console.log(swapcase('Tonight on XYZ-TV'));      // "tONIGHT ON xyz-tv"

File: launchschool_exercises/109_small_problems/easy7/swap_case.rb
def swapcase(str)
  str.split('').map do |ch|
    if ch =~ /[A-Z]/
      ch.downcase
    elsif ch =~ /[a-z]/
      ch.upcase
    else
      ch
    end
  end.join
end
p swapcase('CamelCase') # == 'cAMELcASE'
p swapcase('Tonight on XYZ-TV') # == 'tONIGHT ON xyz-tv'
